OPHD/States/MapViewStateIO.cpp

```diff
@@ -403,6 +403,20 @@ void MapViewState::readStructures(Xml::XmlElement* element)
 			static_cast<SeedLander*>(st)->position({x, y});
 		}
 
+		if (type_id == StructureID::SID_AGRIDOME)
+		{
+			auto agridome = static_cast<Agridome*>(st);
+
+			auto foodStorage = structure->firstChildElement("food");
+			if (foodStorage == nullptr)
+			{
+				throw std::runtime_error("MapViewState::readStructures(): Agridome saved without a food level node.");
+			}
+
+			auto foodLevel = foodStorage->attribute("level");
+			agridome->foodLevel(std::stoi(foodLevel));
+		}
```

Comment:
I feel like this has more XML tag nesting than is required. This implies a parent context such as:
```
<agridome>
  <food level="0" />
</agridome>
```

It might be simpler to have it as:
```
<agridome food="0" />
```

Though I am aware the parent tag might not be named "agridome", and I saw elsewhere you were planning to use numeric structure type IDs.

Reply:
I agree. I want to make that a separate PR and just include food level as an attribute of structures without making it an actual element with a single attribute. Kinda like what the factories do where they simply attach `production_completed` and `production_type` type attributes to a structure node:

```xml
<structure x="164" y="31" depth="1" age="31" state="1" forced_idle="0" disabled_reason="0" idle_reason="0" type="Residential Facility" direction="1" pop0="0" pop1="0" /> <!-- standard structure -->
<structure x="164" y="33" depth="0" age="58" state="1" forced_idle="0" disabled_reason="0" idle_reason="0" type="SEED Factory" direction="1" pop0="2" pop1="0" production_completed="3" production_type="2" /> <!-- factory -->
<structure x="168" y="31" depth="0" age="50" state="1" forced_idle="0" disabled_reason="0" idle_reason="0" type="Agricultural Dome" direction="1" pop0="1" pop1="0" food_level="0" /> <!-- simplified agridome -->
<structure x="164" y="35" depth="0" age="9" state="1" forced_idle="0" disabled_reason="0" idle_reason="0" type="Robot Command" direction="1" pop0="4" pop1="0" robots="4,5,6" /><!-- simplified RCC -->
```

I noticed looking at a savegame that we could also simplify RCC's in the same way. Structures that actually use a storage and production pool makes sense to have the nested elements for each 'pool' and attributes for the associated resources.